app/src/main/java/org/kiwix/kiwixmobile/local_file_transfer/WifiDirectManager.kt

```diff
@@ -0,0 +1,424 @@
+    : Channel? = null
+  private var receiver: BroadcastReceiver? =
+    null // For receiving the broadcasts given by above filter
+  private var groupInfo // Corresponds to P2P group formed between the two devices
+    : WifiP2pInfo? = null
+  private var senderSelectedPeerDevice: WifiP2pDevice? = null
+  private var peerGroupHandshakeAsyncTask: PeerGroupHandshakeAsyncTask? = null
+  private var senderDeviceAsyncTask: SenderDeviceAsyncTask? = null
+  private var receiverDeviceAsyncTask: ReceiverDeviceAsyncTask? = null
+  private var selectedPeerDeviceInetAddress: InetAddress? = null
+  private var fileReceiverDeviceAddress // IP address of the file receiving device
+    : InetAddress? = null
+  private var filesForTransfer: ArrayList<FileItem>? = null
+  var isFileSender = false // Whether the device is the file sender or not
+    private set
+  private var hasSenderStartedConnection = false
+  /* Initialisations for using the WiFi P2P API */
+  fun startWifiDirectManager(filesForTransfer: ArrayList<FileItem>?) {
+    this.filesForTransfer = filesForTransfer
+    isFileSender = filesForTransfer != null && filesForTransfer.size > 0
+    manager = activity.getSystemService(Context.WIFI_P2P_SERVICE) as WifiP2pManager
+    channel = manager!!.initialize(activity, Looper.getMainLooper(), null)
+    registerWifiDirectBroadcastReceiver()
+  }
+
+  private fun registerWifiDirectBroadcastReceiver() {
+    receiver = KiwixWifiP2pBroadcastReceiver(this)
+    // For specifying broadcasts (of the P2P API) that the module needs to respond to
+    val intentFilter = IntentFilter()
+    intentFilter.addAction(WifiP2pManager.WIFI_P2P_STATE_CHANGED_ACTION)
+    intentFilter.addAction(WifiP2pManager.WIFI_P2P_PEERS_CHANGED_ACTION)
+    intentFilter.addAction(WifiP2pManager.WIFI_P2P_CONNECTION_CHANGED_ACTION)
+    intentFilter.addAction(WifiP2pManager.WIFI_P2P_THIS_DEVICE_CHANGED_ACTION)
+    activity.registerReceiver(receiver, intentFilter)
+  }
+
+  private fun unregisterWifiDirectBroadcastReceiver() {
+    activity.unregisterReceiver(receiver)
+  }
+
+  fun discoverPeerDevices() {
+    manager!!.discoverPeers(channel, object : ActionListener {
+      override fun onSuccess() {
+        displayToast(string.discovery_initiated, Toast.LENGTH_SHORT)
+      }
+
+      override fun onFailure(reason: Int) {
+        val errorMessage = getErrorMessage(reason)
+        Log.d(
+          TAG,
+          activity.getString(string.discovery_failed) + ": " + errorMessage
+        )
+        displayToast(string.discovery_failed, Toast.LENGTH_SHORT)
+      }
+    })
+  }
+
+  /* From KiwixWifiP2pBroadcastReceiver.P2pEventListener callback-interface*/
+  override fun onWifiP2pStateChanged(isEnabled: Boolean) {
+    isWifiP2pEnabled = isEnabled
+    if (!isWifiP2pEnabled) {
+      displayToast(string.discovery_needs_wifi, Toast.LENGTH_SHORT)
+      callbacks.onConnectionToPeersLost()
+    }
+    Log.d(
+      TAG,
+      "WiFi P2P state changed - $isWifiP2pEnabled"
```

Comment:
Can be in a single line

Reply:
@rohanprasadofficial I echo this sentiment, a lot of the logs have been spread over too many lines